src/helpers/rangy.ts

```diff
@@ -11,17 +11,53 @@ if (!rangy.initialized) {
 
 export default rangy;
 
+/*
+* Checks whether two RangyRanges share a common ancestor in the DOM tree.
+* This is necessary because certain Rangy operations (like intersectsRange)
+* will throw an error if the ranges do not share a common ancestor node.
+* Using this check before calling intersectsRange prevents runtime exceptions.
+*/
+const haveCommonAncestor = (range1: RangyRange, range2: RangyRange): boolean => {
+  let containerA: typeof range1.commonAncestorContainer | null =
+      range1.commonAncestorContainer;
+    const containerB: typeof range2.commonAncestorContainer | null =
+      range2.commonAncestorContainer;
+    while (containerA) {
+      let current: typeof containerB | null = containerB;
+      while (current) {
+        if (containerA === current) return true;
+        current = current.parentNode;
+      }
+      containerA = containerA.parentNode;
+    }
+    return false;
```

Comment:
How would they not share an ancestor? Would they need to be in different DOMs, or detached? Otherwise the body etc should be a common ancestor right?

When I tried the search in my browser, I only saw the error in `findText`, which is already handled by the try/catch you added.